spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -20,33 +20,141 @@
+image::overall.png[Overall architecture]
 
-=== Design
+Application's beans contain methods annotated with `@Incoming` and `@Outgoing` annotations. 
+A method with an `@Incoming` annotation consumes messages from a _channel_. 
+A method with an `@Outgoing` annotation produces messages, it emits messages to a _channel_.
+A method with both an `@Incoming` and an `@Outgoing` annotation is a message processor, it consumes messages from a _channel_, does some transformation to them, and emits messages to another_channel_.
+
+==== Channel
+
+A _channel_ is a name indicating from expected messages are consumed or to where messages are sent.
+Channels are opaque `Strings`.
+
+There are two type of channel:
+
+* Internal channels are local to the application. 
+It allows implementing multi-step processing where several _beans_ from the same application form a chain of processing.
+* Channels can be _connected_ to remote brokers or various message transport layer such as Apache Kafka or to an AMQP broker.
+These channels are managed by _connectors_.
+
+==== Message
+
+At the core of the Reactive Messaging specification is the concept of _message_. 
+A _message_  is an enveloppe wrapping a _payload_. 
+A message is sent to a specific channel and, when received and processed successfuly, _acknowledged_.
+
+Reactive Messaging application components are addressable recipients await the arrival of messages on a channel and react to them, otherwise lying dormant. 
+
+Message are represented by the `org.eclipse.microprofile.reactive.messaging.Message` class. 
+This interface is kept minimal on purpose aiming _connectors_ to provide their own with additional metadata.
+For instance, a `KafkaMessage` would provide access to the _topic_ and _partition_.
+
+The `org.eclipse.microprofile.reactive.messaging.Message#getPayload` method retrieves the wrapped payload. 
+The `org.eclipse.microprofile.reactive.messaging.Message#ack` method acknowledges the message.
+Note that the `ack` method is asynchronous as acknowledgement is generally an asynchronous process.
+
+_Plain_ messages are created using:
+
+* `org.eclipse.microprofile.reactive.messaging.Message#of(T)` - wraps the given payload, no acknowledgement
+* `org.eclipse.microprofile.reactive.messaging.Message#of(T, java.util.function.Supplier<java.util.concurrent.CompletionStage<java.lang.Void>>)` - wraps the given payload and provides the acknowlegment logic
+
+==== Message consumption with @Incoming
+
+The `org.eclipse.microprofile.reactive.messaging.Incoming` annotation is used on a method from a CDI bean to indicate that the method consumes messages from the specified channel:
+
+[source,java]
+----
+@Incoming("my-channel")                                                // <1>
+public CompletionStage<Void> consume(Message<String> message) {        // <2>
+  return message.ack();
+}
+----
+1. `my-channel` is the channel
+2. the method is called for every message sent to the `my-channel` channel
+
```

Comment:
Would it be worth introducing how the specification intends implementation to do flow control here? Are there circumstances where the method might not be called....yet...as the system is creating some form of balancing 'back-pressure'?

Should we acknowledge the concept that some messages sent to an external channel to
be received by (only) a single @Incoming bean in (only one of) the consuming services?
Or is that a matter completely for the semantics of the messaging provider?